### DISCUSSION TITLE
required setting is unnecessary for lightswitch field

### DISCUSSION THREAD

**Discussion by michel-o** [craftcms/cms#8568] [Ideas]
It can only be `on` or `off`, there is no "empty" value.

**Comment by brandonkelly** [comment]
Correct, but the field layout designer doesn’t know that. I’ll consider this a feature request.

**Comment by billythekid** [comment]
Would required be true if the lightswitch was set to `off`? I always thought setting required on a lightswitch meant it was required to be turned `on` to pass validation. Used in cases such as, confirm you agree to the terms before saving, etc. (in guest entries for example)

**Comment by brandonkelly** [comment]
@billythekid No, it’s impossible to get a Required validation error for a Lightswitch field.

If you wanted a mandatory checkbox thing, you could do it with a required Checkboxes field, with one option.
